src/libs/filters/rating.c

```diff
@@ -99,21 +99,19 @@ static gchar *_rating_print_func(const double value, const gboolean detailled)
         return g_strdup(_("rejected"));
       case 0:
         return g_strdup(_("not rated"));
-      case 1:
-        return g_strdup("★");
-      case 2:
-        return g_strdup("★ ★");
-      case 3:
-        return g_strdup("★ ★ ★");
-      case 4:
-        return g_strdup("★ ★ ★ ★");
-      case 5:
-        return g_strdup("★ ★ ★ ★ ★");
     }
   }
   return g_strdup_printf("%.0lf", floor(value));
 }
 
+static gchar *_rating_current_text_func(GtkDarktableRangeSelect *range, const double current)
+{
+  return g_strdup_printf("  <b>%s</b> | %s: %s  ", 
+        g_markup_escape_text(range->print(current, TRUE), -1),
+        _("selected"),
+        g_markup_escape_text(dtgtk_range_select_get_rating_bounds_pretty(range), -1));
```

Comment:
and if it does indeed create a duplicate string here we have a memory leak.

Reply:
you're right indeed !
@quovadit : either you fix it, or I will do that once your PR is merged !

Reply:
I'm really sorry, but I have no idea what's wrong, so unfortunately I can't fix it

Reply:
`range->print` return an already duplicated string. So we need to free it after use. Something like : 
```
gchar *txt = range->print(current, TRUE);
gchar *txt2 = g_strdup_printf("  <b>%s</b> | %s: %s  ", 
        g_markup_escape_text(txt, -1),
        _("selected"),
        g_markup_escape_text(dtgtk_range_select_get_rating_bounds_pretty(range), -1));
g_free(txt);
return txt2;
```

Reply:
I hope I fixed it correctly...

Reply:
yes... but I've forgotten to free the result of `g_markup_escape_text` (see below) Sorry for that !